lib/IMAP/Sync/Synchronizer.php

```diff
@@ -86,7 +85,51 @@ public function sync(Horde_Imap_Client_Base $imapClient,
 		$changedMessages = $this->messageMapper->findByIds($imapClient, $request->getMailbox(), $changedUids, $userId);
 		$vanishedMessageUids = $vanishedUids;
 
-		return new Response($newMessages, $changedMessages, $vanishedMessageUids, null);
+		$this->requestId = $request->getId();
+		$this->response = new Response($newMessages, $changedMessages, $vanishedMessageUids, null);
+		return $this->response;
+	}
+
+	/**
+	 * @return array [$newUids, $changedUids, $vanishedUids]
+	 * @throws Horde_Imap_Client_Exception
+	 * @throws Horde_Imap_Client_Exception_Sync
+	 */
+	private function doCombinedSync(Horde_Imap_Client_Base $imapClient, Horde_Imap_Client_Mailbox $mailbox, Request $request): array {
+		$syncData = $imapClient->sync($mailbox, $request->getToken(), [
+			'criteria' => Horde_Imap_Client::SYNC_ALL,
+		]);
+
+		return [
+			$syncData->newmsgsuids->ids,
+			$syncData->flagsuids->ids,
+			$syncData->vanisheduids->ids,
+		];
+	}
+
+	/**
+	 * @return array [$newUids, $changedUids, $vanishedUids]
```

Comment:
is this a format Psalm understands? I've only seen it like ``array<0: int[], 1: int[], 2: int[]>``

Reply:
There is also a short form of this, the `list` type[^1] for annotating tuples. The problem was that Horde is missing typings (for `flagsuids->ids` etc.) and thus Psalm complained.

I'll fix it.

[^1]: https://psalm.dev/docs/annotating_code/type_syntax/array_types/#list-shapes